From 0764674a19337a63044fc80cbd777430bbb9f845 Mon Sep 17 00:00:00 2001
From: Xiang Wang <xiang.a.wang@intel.com>
Date: Thu, 14 May 2015 10:43:04 +0800
Subject: [PATCH] REVERTME: i2c: designware: move i2c suspend/resume sequence

Change suspend/resume callbacks to suspend_late/resume_early.
So that it's safe to access i2c in all other drivers' suspend/resume
callbacks.

Change-Id: I9c6babbc5d9b374f3baa6e480aced568b2ac0b70
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-26433
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1557
Signed-off-by: Xiang Wang <xiang.a.wang@intel.com>
---
 drivers/i2c/busses/i2c-designware-platdrv.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-designware-platdrv.c b/drivers/i2c/busses/i2c-designware-platdrv.c
index 6f9f92405b7a..105e28af3ae3 100644
--- a/drivers/i2c/busses/i2c-designware-platdrv.c
+++ b/drivers/i2c/busses/i2c-designware-platdrv.c
@@ -434,7 +434,7 @@ MODULE_DEVICE_TABLE(of, dw_i2c_of_match);
 #endif
 
 #ifdef CONFIG_PM
-static int dw_i2c_suspend(struct device *dev)
+static int dw_i2c_suspend_late(struct device *dev)
 {
 	struct platform_device *pdev = to_platform_device(dev);
 	struct dw_i2c_dev *i_dev = platform_get_drvdata(pdev);
@@ -454,7 +454,7 @@ static int dw_i2c_suspend(struct device *dev)
 	return 0;
 }
 
-static int dw_i2c_resume(struct device *dev)
+static int dw_i2c_resume_early(struct device *dev)
 {
 	struct platform_device *pdev = to_platform_device(dev);
 	struct dw_i2c_dev *i_dev = platform_get_drvdata(pdev);
@@ -505,7 +505,10 @@ static int dw_i2c_runtime_resume(struct device *dev)
 #endif
 
 static const struct dev_pm_ops dw_i2c_dev_pm_ops = {
-	SET_SYSTEM_SLEEP_PM_OPS(dw_i2c_suspend, dw_i2c_resume)
+#ifdef CONFIG_PM_SLEEP
+	.suspend_late = dw_i2c_suspend_late,
+	.resume_early = dw_i2c_resume_early,
+#endif
 	SET_RUNTIME_PM_OPS(dw_i2c_runtime_suspend, dw_i2c_runtime_resume, NULL)
 };
 /* work with hotplug and coldplug */
-- 
1.9.1

